utils/errors/errors.go

```diff
@@ -77,10 +77,12 @@ func (c customError) Cause() error {
 }
 
 func (c customError) GRPCStatus() *grpcstatus.Status {
-	if c.status == nil {
-		return grpcstatus.New(codes.Internal, c.Error())
+	statusCode := codes.Internal
+	if c.status != nil {
+		statusCode = c.status.Code()
 	}
-	return c.status
+
+	return grpcstatus.New(statusCode, c.Error()) // always return with latest error message
```

Comment:
If `c.status` contains any details, the details will be lost when you create a new grpcstatus.Status struct.

Reply:
right, I'll update it